src/Data/URI/Host/Gen.purs

```diff
@@ -0,0 +1,15 @@
+module Data.URI.Host.Gen where
+
+import Prelude
+
+import Control.Monad.Gen as Gen
+import Data.String as S
+import Data.URI.Host (Host(..))
+
+genIPv4 :: forall m. Gen.MonadGen m => m Host
+genIPv4 = do
```

Comment:
Is that a valid notation for IPv4 addresses? I haven't seen them formatted that way before.

I think the parser will work for that actually, so should either improve the generator or make the parser stricter I guess!